src/restconfig/src/main/java/org/geoserver/rest/catalog/RestConfigurationLockCallback.java

```diff
@@ -43,7 +43,8 @@ public void dispatched(
             HttpServletRequest request, HttpServletResponse response, Object handler) {
         Object controller = DispatcherCallback.getControllerBean(handler);
         if (controller instanceof AbstractCatalogController
-                || controller instanceof AbstractGeoServerController) {
+                || controller instanceof AbstractGeoServerController
+                || controller instanceof SequentialExecutionController) {
             if (controller instanceof CatalogReloadController
                     || isWriteMethod(request.getMethod())) {
                 // this requires a full lock, it affects part of GeoTools that are not thread safe
```

Comment:
This comment probably should be updated too. Perhaps something like:

```suggestion
                // this requires a full lock, it affects parts of GeoServer or GeoTools that are not thread safe
```